src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/ObjectWriter/Dwarf/DwarfFde.cs

```diff
@@ -0,0 +1,113 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics;
+using System.Buffers;
+using static ILCompiler.ObjectWriter.DwarfNative;
+
+namespace ILCompiler.ObjectWriter
+{
+    internal sealed class DwarfFde
+    {
+        public readonly DwarfCie Cie;
+        public string PcStartSymbolName;
+        public long PcStartSymbolOffset;
+        public ulong PcLength;
+        public string LsdaSymbolName;
+        public string PersonalitySymbolName;
+        public readonly byte[] Instructions;
+
+        public DwarfFde(DwarfCie cie, byte[] instructions)
+        {
+            Cie = cie;
+            Instructions = instructions;
+            PersonalitySymbolName = null;
+        }
+
+        private enum CFI_OPCODE
+        {
+            CFI_ADJUST_CFA_OFFSET,    // Offset is adjusted relative to the current one.
+            CFI_DEF_CFA_REGISTER,     // New register is used to compute CFA
+            CFI_REL_OFFSET,           // Register is saved at offset from the current CFA
+            CFI_DEF_CFA               // Take address from register and add offset to it.
+        }
+
+        /// <summary>
+        /// Convert JIT version of CFI blob into the the DWARF byte code form.
+        /// </summary>
+        public static byte[] CfiCodeToInstructions(DwarfCie cie, byte[] blobData)
+        {
+            int cfaOffset = cie.InitialCFAOffset;
+            var cfiCode = ArrayPool<byte>.Shared.Rent(4096);
+            int cfiCodeOffset = 0;
+            byte codeOffset = 0;
+            byte lastCodeOffset = 0;
+            int offset = 0;
+            while (offset < blobData.Length)
+            {
+                codeOffset = Math.Max(codeOffset, blobData[offset++]);
+                CFI_OPCODE opcode = (CFI_OPCODE)blobData[offset++];
+                short dwarfReg = BitConverter.ToInt16(blobData, offset);
+                offset += sizeof(short);
+                int cfiOffset = BitConverter.ToInt32(blobData, offset);
+                offset += sizeof(int);
+
+                if (codeOffset != lastCodeOffset)
+                {
+                    // Advance
+                    int diff = (int)((codeOffset - lastCodeOffset) / cie.CodeAlignFactor);
+                    if (diff <= 0x3F)
+                    {
+                        cfiCode[cfiCodeOffset++] = (byte)((byte)DW_CFA_advance_loc | diff);
+                    }
```

Comment:
`DW_CFA_..` are `byte` constants. No need to cast.

Reply:
That's likely left over from the time I used `enum` for those. I'll try to remove it.